src/pkcs11/framework-pkcs15.c

```diff
@@ -2817,25 +2866,38 @@ pkcs15_create_public_key(struct sc_pkcs11_slot *slot, struct sc_profile *profile
 		}
 	}
 
+	if (key_type == CKK_EC_EDWARDS) {
+		if (ec->ecpointQ.len == 32)
+			ec->params.id = oid_ED25519;
+		else
+			ec->params.id = oid_ED448;
+	} else if (key_type == CKK_EC_MONTGOMERY) {
+		if (ec->ecpointQ.len == 32)
+			ec->params.id = oid_X25519;
+		else
+			ec->params.id = oid_X448;
```

Comment:
I think setting the OID just based on the else branch is quite error prone -- there should be explicit check for the right length and then we should fail if we receive different length to avoid processing trash. Same for ED above

Reply:
Will test for 56 byte  for ED448 and X448.